src/presentation/uiutils.ts

```diff
@@ -1,52 +1,100 @@
 export const myAutoresizeOptions = { padding: 2, minWidth: 10, maxWidth: 800 };
 
-export function installAutoresize(textWidthAlternativeCalculator?: (text: string) => number): void {
-  // @ts-ignore
-  $.fn.textWidth = function (_text: string, _font) {
-    // get width of text with font.  usage: $("div").textWidth();
-    let textToConsider = _text || (this.val() as string);
-    if (textToConsider === '') {
-      textToConsider = (this[0] as HTMLInputElement).placeholder;
+function textWidth(elOrText?: HTMLElement | string, options: InputWidthOptions = {}): number {
+  // get width of text with font.
+  let textToConsider = (typeof(elOrText) === 'string') ? elOrText : ((elOrText as any).value as string);
+  if (textToConsider === '' && (elOrText instanceof window.HTMLInputElement)) {
+    textToConsider = elOrText.placeholder;
+  }
+  const padding = options.padding || 0;
+  if (options.widthCalculator) {
+    return options.widthCalculator(textToConsider) + padding;
+  }
+  if(!document.body) {
+    return 0;
+  }
+  const fakeEl = document.createElement('span');
+  fakeEl.style.visibility = 'hidden';
+  document.body.appendChild(fakeEl);
+  fakeEl.innerText = textToConsider;
+  const font = options.font || (elOrText as any).style?.font;
+  if(font) {
+    fakeEl.style.font = font;
+  }
+  fakeEl.style.whiteSpace = 'pre';
+  const width = fakeEl.getBoundingClientRect().width;
+  fakeEl.remove();
+  return width + padding;
+}
+
+export type InputWidthOptions = {
+  padding?: number;
+  minWidth?: number;
+  maxWidth?: number;
+  widthCalculator?: (text: string) => number
+  font?: string;
+};
+export function inputWidthUpdate(el: HTMLElement, options?: InputWidthOptions): void {
+  options = {...{ padding: 10, minWidth: 0, maxWidth: 10000 }, ...(options || {})};
+
+  el.style.width = `${Math.min(
+      options.maxWidth || 0,
+      // @ts-ignore
+      Math.max(options.minWidth, textWidth(el, options)),
+  )}px`;
+}
+
+export function autoresize(el: HTMLElement, options?: InputWidthOptions) {
+  // resizes elements based on content size.  usage: autoresize(someInput, {padding:10,minWidth:0,maxWidth:100});
+  const eventType = 'input';
```

Comment:
Does this work better? Does this work both on Chrome and Firefox?

Reply:
It's the same code as before, only repackaged without jQuery. It still doesn't work quite right on my machine, both on Firefox and on Chrome. It sizes the inputs too small.